packages/@lwc/integration-karma/test/component/LightningElement.errorCallback/index.spec.js

```diff
@@ -205,3 +222,156 @@ describe('error boundary during initial component construction', () => {
+            e.preventDefault();
+            caughtError = e.reason;
+        };
+
+        beforeEach(() => {
+            // Overriding window.onerror disables Jasmine's global error handler, so we can listen for errors
+            // ourselves. There doesn't seem to be a better way to disable Jasmine's behavior here.
+            // https://github.com/jasmine/jasmine/pull/1860
+            originalOnError = window.onerror;
+            // Dummy onError because Jasmine tries to call it in case of a rejection:
+            // https://github.com/jasmine/jasmine/blob/169a2a8/src/core/GlobalErrors.js#L104-L106
+            window.onerror = () => {};
+            caughtError = undefined;
+            window.addEventListener('unhandledrejection', onRejection);
+        });
+
+        afterEach(() => {
+            window.removeEventListener('unhandledrejection', onRejection);
+            window.onerror = originalOnError;
+        });
+
+        function testStub(testcase, hostSelector, hostClass) {
+            it(`parent errorCallback throws after value mutation ${testcase}`, () => {
+                const throwElm = createElement(hostSelector, { is: hostClass });
+                const noThrowElm = createElement('x-no-throw-on-mutate', { is: XNoThrowOnMutate });
+                document.body.appendChild(throwElm);
+                document.body.appendChild(noThrowElm);
+                return (
+                    Promise.resolve()
+                        .then(() => {
+                            throwElm.show = true;
+                            noThrowElm.show = true;
+                        })
+                        // Need to wait a few ticks so flushRehydrationQueue can finish
+                        .then(() => new Promise((resolve) => setTimeout(resolve)))
+                        .then(() => new Promise((resolve) => setTimeout(resolve)))
+                        .then(() => {
+                            // error is thrown by parent's errorCallback
+                            expect(caughtError).not.toBeUndefined();
+                            expect(caughtError.message).toMatch(
+                                /error in the parent error callback after value mutation/
+                            );
+                            // child after the throwing child is not rendered
+                            expect(
+                                throwElm.shadowRoot.querySelector('x-after-throwing-child')
+                            ).toBeNull();
+                            // An unrelated element rendered after the throwing parent still renders. I.e. we didn't
+                            // give up rendering entirely just because one element threw in errorCallback.
+                            expect(noThrowElm.shadowRoot.querySelector('div').textContent).toEqual(
+                                'shown'
+                            );
```

Comment:
This test will fail if we remove this code:

https://github.com/salesforce/lwc/blob/61a89c3f41a211bcc27b4878f2f2d83b279e1ea6/packages/@lwc/engine-core/src/framework/vm.ts#L570-L586

Also, this is the only case in our Karma tests where this code gets hit!

Reply:
Nice find, it took me a while to figure out how this worked!

Is it because [`componentValueMutated`](https://github.com/salesforce/lwc/blob/ac988f4d85a02ed5c53093ed771f5c63a565cd41/packages/@lwc/engine-core/src/framework/update-component-value.ts#L10-L17) triggers the [`reactiveObserver`](https://github.com/salesforce/lwc/blob/ac988f4d85a02ed5c53093ed771f5c63a565cd41/packages/@lwc/engine-core/src/framework/component.ts#L46-L54) to run `scheduleRehydration`?

Reply:
Correct, it has to be triggered by a value mutation. This is a special code path that doesn't occur during initial render.